### PR TITLE
feat(exporters/elasticsearch-exporter): adds basic HTTP auth to exporter

### LINKED ISSUE #1984: Elasticsearch Exporter supports basic HTTP authentication
AT:

- I can configure the exporter with a username/password
- The exporter can connect to a secured Elasticsearch cluster through the REST API with basic HTTP authentication

### PR DESCRIPTION
**Pull request by npepinpe**
- adds basic auth config for username/password

closes #1984

### CODE REVIEW

**Comment by npepinpe** [comment]
Let's discuss how we're testing this

**Comment by npepinpe** [comment]
I did a bit more digging into how we could test ES, and I think TestContainers is probably the "best" in this case:

1. Embedded using `ESIntegTest` is limited and is designed for Elastic's own test suite, and is not their recommended way that third-parties should do integration tests, which makes sense.
1. Using the Maven plugin means we can't configure the cluster on a per-test or per-suite instance, but need more complicated setup when running integration tests.
1. Using [a forked JVM](https://github.com/allegro/embedded-elasticsearch) seems fine in general, but it relies on their setup around how to start and configure Elastic, which needs to be updated by them on every new version, tested, etc.

`TestContainers` is really just a thin wrapper around the official ES docker images, which already support different way of configuring the instances, and comes built in with picking a random port. Considering the other options, this is probably our "best" bet, even if it means a hard dependency on docker for running the tests. We kind of already have a dependency on docker since it's part of our getting started guide for developers, so...

**Comment by menski** [comment]
This is blocked by CI as it now requires docker for the test container. We could migrate to the docker in docker image but then we got the problem that the test container uses a newer docker version then we have installed on your HQ cluster. Possible ideas, go with a non docker solution, check if we can specify the docker version used by test containers, or migrate to the cloud.

**Comment by npepinpe** [comment]
As per https://www.testcontainers.org/supported_docker_environment/

Required Docker version for dind is 1.12. I'm not sure what version we have but I'm gonna assume it doesn't matter if we can specify it or not =/

I still think the containers were a better alternative than the forked JVM, but we can't really wait until we migrate CI to the cloud.

**Comment by npepinpe** [comment]
I ended up going with the "embedded" solution (the forked vm), which is sort of ok; downsides is it's a lot harder to customize (e.g. it was a pain to setup authentication because it's not really meant to work with that...), so I would still hope we will find a solution to use TestContainers in the future. In the meantime there's an interface `ElasticsearchNode`, which should help us in the future migrate to `TestContainers` painlessly.

File: exporters/elasticsearch-exporter/pom.xml
Comment (menski) [inline exporters/elasticsearch-exporter/pom.xml]: remove the plugin also from the parent pom as we are not using it anymore

File: exporters/elasticsearch-exporter/src/main/java/io/zeebe/exporter/ElasticsearchExporterConfiguration.java
Comment (menski) [inline exporters/elasticsearch-exporter/src/main/java/io/zeebe/exporter/ElasticsearchExporterConfiguration.java]: this has to be documented in our default configuration otherwise a user will never find it, add it here https://github.com/zeebe-io/zeebe/blob/1984-es-auth/dist/src/main/config/zeebe.cfg.toml#L319-L347

File: exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/util/ElasticsearchForkedJvm.java
Comment (menski) [inline exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/util/ElasticsearchForkedJvm.java]: This will most likely become out of sync once dependabot bumps the next elastic version. I see two options at the moment and favor the second one:

1. set a system property from maven to access the `${version.elasticsearch}` maven property https://github.com/zeebe-io/zeebe/blob/9ca84c9fa756c67333aa1598ae1a73356b196f03/parent/pom.xml#L33
1. fetch the version from the java metadata, i.e.
    ```
    ElasticsearchClient.class.getPackage().getImplementationVersion() 
    ```

**Review by menski — CHANGES_REQUESTED** [review]
LGTM just some small remarks

**Comment by menski** [comment]
Another thing I just noticed, at the moment we print the whole configuration on startup which now includes the elasticsearch password. Do think this is a problem?

**Comment by npepinpe** [comment]
Does it include it? I thought I'd left it out of the `#toString` method

**Comment by menski** [comment]
The config printing isn't using the `toString` method, its just gson pretty printing the configuration class.

https://github.com/zeebe-io/zeebe/blob/9ca84c9fa756c67333aa1598ae1a73356b196f03/broker-core/src/main/java/io/zeebe/broker/system/configuration/BrokerCfg.java#L151

**Comment by npepinpe** [comment]
Ah, then in that case yes I'd say that's an issue. Setting it to `transient` should solve this, as Gson will ignore it.

**Comment by npepinpe** [comment]
I updated based on your feedback, and about the password I create #2019.

**Review by menski — APPROVED** [review]

**Comment by menski** [comment]
Please remove the system property solution, as the implementation version is enough I think. After that feel free to squash and merge

**Comment by npepinpe** [comment]
bors retry

**Pull request merged**
